enumerable.rb

```diff
@@ -0,0 +1,126 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum :my_each unless block_given?
+
+    for i in self
+      yield(i)
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum :my_each unless block_given?
+
+    for i in 0...size
+      yield(to_a[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum :my_select unless block_given?
+
+    if is_a? Hash
+      helper = {}
+      my_each { |k, v| helper[k] = v if yield(k, v) }
+    else
+      helper = []
+      my_each { |i| helper << i if yield(i) }
+    end
+    helper
+  end
+
+  def my_all?(args = nil)
+    query_result = true
+    if args.nil? && !block_given?
+      my_each { |i| query_result = false if !i || i.nil? }
+    elsif args.nil?
+      my_each { |i| query_result = false unless yield(i) }
+    elsif args.is_a? Regexp
+      my_each { |i| query_result = false unless i.match(args) }
+    elsif args.is_a? Module
+      my_each { |i| query_result = false unless i.is_a?(args) }
+    else
+      my_each { |i| query_result = false unless i == args }
+    end
+    query_result
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    count = size if !block_given? && arg.nil?
+    my_each { |i| count += 1 unless i != arg } if arg
+    my_each { |i| count += 1 if yield(i) } if block_given?
+    puts count
+  end
+
+  def my_any?(arg = nil)
+    query_result = false
+    if !block_given? && arg.nil?
+      my_each { |i| query_result = true unless i.nil? || !i }
+    elsif arg.nil?
+      query_result = true if count { |i| yield(i) }.positive?
+    elsif arg.is_a? Regexp
+      my_each { |i| query_result = true if i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| query_result = true if i.is_a?(arg) }
+    else
+      count(arg).positive? ? query_result = true : next
+    end
+    query_result
+  end
+
+  def my_none?(arg = nil)
+    query_result = true
+    if !block_given? && arg.nil?
+      my_each { |i| query_result = false unless i.nil? || !i }
+    elsif arg.nil?
+      my_each { |i| query_result = false if yield(i) }
+    elsif arg.is_a? Regexp
+      my_each { |i| query_result = false if i.match(arg) }
+    elsif arg.is_a? Module
+      my_each { |i| query_result = false if i.is_a?(arg) }
+    else
+      my_each { |i| query_result = false if i == arg }
+    end
+    query_result
+  end
+
+  def my_map
+    return to_enum :my_map unless block_given?
+
+    ary = []
+    my_each { |i| ary << yield(i) }
+    puts ary
+  end
+
+  def my_map_proc(proc = nil)
+    return to_enum :my_map_proc unless block_given?
+
+    ary = []
+    my_each { |i| ary << proc ? proc.call(i) : yield(i) }
+    ary
+  end
+
+  def my_inject(arg = nil)
+    if !arg && block_given?
+      memo = self[0]
+      first_memo = self[1..-1]
+    else
+      memo = arg
+      first_memo = self
+    end
+    first_memo.my_each do |i|
+      memo = yield(memo, i) if block_given?
+    end
+    puts memo
+  end
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method should work with symbols as arguments (:+) and blocks and perform the right operation. Right now it's returning nil.

Reply:
solved, thanks for the correction